sql/ttl: extract TTL query bounds logic for general reuse

spilchen opened the pull request:
Previously, the row-level TTL implementation contained logic to convert a span into WHERE clause predicates. This logic is now needed for the new INSPECT feature, but it resides within the ttl package, which we don't want to import into the inspect code.

This change refactors the relevant code into a new package, sql/spanutils, allowing it to be reused by INSPECT without adding the ttl dependency.

Closes #148294

Epic: CRDB-30356
Release note: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/149048)
<!-- Reviewable:end -->

spilchen committed:
sql/spanutils: extract TTL query bounds logic for general reuse

Previously, the row-level TTL implementation contained logic to convert a span
into WHERE clause predicates. This logic is now needed for the new INSPECT
feature, but it resides within the ttl package, which we don't want to import
into the inspect code.

This change refactors the relevant code into a new package, sql/spanutils,
allowing it to be reused by INSPECT without adding the ttl dependency.

Closes #148294

Epic: CRDB-30356
Release note: none

rafiss reviewed (approved):
lgtm! this will be a useful package 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/149048#-:-OTmYe1b2Bt2kzRQCyYj:b-3de79w)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @spilchen)



<!-- Sent from Reviewable.io -->

spilchen commented:
TFTR!

bors r+
